postgresql/database.go

```diff
@@ -87,8 +87,13 @@ func (d *database) WithSession(sess interface{}) (db.Database, error) {
 		clone.tx = sqltx.New(tx)
 	}
 
-	clone.cachedStatements = cache.NewCache()
-	clone.collections = make(map[string]*table)
+	d.collectionsMu.Lock()
```

Comment:
I'd actually think of having Ctrie (lock-free hash array mapped trie with efficient non-blocking snapshots) here instead of map+lock.